taichi/ir/frontend_ir.cpp

```diff
@@ -688,6 +688,39 @@ void MatrixExpression::flatten(FlattenContext *ctx) {
   stmt->ret_type = this->dt;
 }
 
+IndexExpression::IndexExpression(ASTBuilder *builder,
+                                 const Expr &var,
+                                 const ExprGroup &indices,
+                                 std::string tb)
+    : var(var) {
+  // IndexExpression without ret_shape is used for matrix indexing,
+  // where each entry of ExprGroup is interpreted as indexing into a specific
+  // axis. For example, mat[0][3, 4] has indices_group={0, [3, 4]}, where [3, 4]
+  // corresponds to "n"-axis and "m"-axis of the matrix. Therefore we expand
+  // indices_group={0, [3, 4]} into {0, 3, 4} to avoid TensorType in indices.
```

Comment:
The comment confuses me a bit. Why does a matrix need three indices `0, 3, 4`?

Reply:
Ah bad example, let me change it to {3, 4}